calyx-py/calyx/queue_util.py

```diff
@@ -12,14 +12,33 @@ def parse_json():
     data = json.load(sys.stdin)
     commands = data["commands"]["data"]
     values = data["values"]["data"]
+
+    if parse_ranks:
+        ranks = data["ranks"]["data"]
+    if parse_times:
+        times = data["times"]["data"]
+
+    if parse_ranks and parse_times:
+        return commands, values, ranks, times
+    if parse_times:
+        return commands, values, times
+    if parse_ranks:
+        return commands, values, ranks
```

Comment:
Feels like it will confuse new users, since calling the same function can return 3 or 4 things, and when 3 things come back they could be of two different species. Consider setting this to be a 4-tuple, where you just return `None` sometimes. Alternately you could return a Python dict so the caller can access the appropriate fields using dot-notation

Reply:
In my updated version, I'm setting it to a 4-tuple and pattern matching the necessary data out inside the `*_oracle.py` files as is necessary. That should reduce the ambiguity of what is returned.